docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,37 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rollingupdate rc foo --image=myimage:v2```
+
+Under the hood, this will do the following:
+   * Create a new replication controller named foo-next, identical to foo, except image is updated to the new image.
+   * Do the standard down/up one-by-one until foo has replicas zero and foo-next has replicas N
+   * delete rc foo
+   * create rc foo that is identical to foo-next
+   * delete foo-next
```

Comment:
What happens if the server dies somewhere during this multi-stage process? Do we require the client to detect this and finish the remaining operations when the server comes back up? Can we use annotations to help with this, like we're currently using annotations with kubectl rollingupdate?